geomesa-spark/geomesa-spark-test-with-sedona/pom.xml

```diff
@@ -0,0 +1,91 @@
+
+    <name>GeoMesa Spark Test With Sedona</name>
+    <artifactId>geomesa-spark-test-with-sedona_2.12</artifactId>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.spark</groupId>
+            <artifactId>spark-core_${scala.binary.version}</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.spark</groupId>
+            <artifactId>spark-sql_${scala.binary.version}</artifactId>
+            <version>${spark.version}</version>
+            <scope>provided</scope>
+        </dependency>
+
+        <!-- test -->
+        <dependency>
+            <groupId>org.specs2</groupId>
+            <artifactId>specs2-core_${scala.binary.version}</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.specs2</groupId>
+            <artifactId>specs2-junit_${scala.binary.version}</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.locationtech.geomesa</groupId>
+            <artifactId>geomesa-spark-sql_${scala.binary.version}</artifactId>
+            <version>${project.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.locationtech.geomesa</groupId>
+            <artifactId>geomesa-gt-spark_${scala.binary.version}</artifactId>
+            <version>${project.version}</version>
+            <scope>test</scope>
+        </dependency>
+
+        <!-- for testing Apache Sedona integration -->
+        <dependency>
+            <groupId>org.wololo</groupId>
+            <artifactId>jts2geojson</artifactId>
+            <version>${jts2geojson.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.sedona</groupId>
+            <artifactId>sedona-core-${spark.compat.version}_${scala.binary.version}</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.sedona</groupId>
+            <artifactId>sedona-sql-${spark.compat.version}_${scala.binary.version}</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>test-jar</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <systemPropertyVariables combine.children="append">
+                        <geomesa.use.sedona>true</geomesa.use.sedona>
```

Comment:
Ah!  You don't have to work this hard to set the system variables.  You can set and unset them in unit test like this:

https://github.com/locationtech/geomesa/blob/main/geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreAuthTest.scala#L82-L87

This would avoid needing to create the separate modules.

Reply:
The reason for placing sedona test in separated module is to ensure that it runs in its own process, since [UDT serde](https://github.com/locationtech/geomesa/pull/2802/files#diff-f7fb90df88946514556bd7ae01e8de1c3a18d4db0a8d05cd37d07a31e2c87f0fR50) only initialize once for performance reason. We can also override `reuseForks` config of surefire plugin in `geomesa-spark-sql` module to achieve this. I'll submit a patch if you prefer this way.

`geomesa-spark-test-without-sedona` have to run in its own module. This is for validating no sedona classes will be loaded when `geomesa.use.sedona` is not enabled, so we want it to run in an environment without sedona on the classpath.